src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.ts

```diff
@@ -71,5 +105,65 @@ export class ExamChecklistComponent implements OnChanges, OnInit, OnDestroy {
         this.websocketService.unsubscribe(submittedTopic);
         const startedTopic = this.examChecklistService.getStartedTopic(this.exam);
         this.websocketService.unsubscribe(startedTopic);
+        if (this.longestWorkingTimeSub) {
+            this.longestWorkingTimeSub.unsubscribe();
+        }
+    }
+
+    /**
+     * Evaluates all the quiz exercises that belong to the exam
+     */
+    evaluateQuizExercises() {
+        this.isEvaluatingQuizExercises = true;
+        if (this.exam.course?.id !== undefined && this.exam.id !== undefined) {
+            this.examManagementService.evaluateQuizExercises(this.exam.course.id, this.exam.id).subscribe({
+                next: (res) => {
+                    this.alertService.success('artemisApp.studentExams.evaluateQuizExerciseSuccess', { number: res?.body });
+                    this.existsUnassessedQuizzes = false;
+                    this.isEvaluatingQuizExercises = false;
+                },
+                error: (error: HttpErrorResponse) => {
+                    this.dialogErrorSource.next(error.message);
+                    this.alertService.error('artemisApp.studentExams.evaluateQuizExerciseFailure');
+                    this.isEvaluatingQuizExercises = false;
+                },
+            });
+        } else {
+            this.alertService.error('artemisApp.studentExams.evaluateQuizExerciseIdFailure');
+        }
+    }
+
+    /**
+     * Evaluates all the unsubmitted Text and Modelling submissions to 0
+     */
+    assessUnsubmittedExamModelingAndTextParticipations() {
+        this.isAssessingUnsubmittedExams = true;
+        if (this.exam.course?.id !== undefined && this.exam.id !== undefined) {
+            this.examManagementService.assessUnsubmittedExamModelingAndTextParticipations(this.exam.course.id, this.exam.id).subscribe({
+                next: (res) => {
+                    this.alertService.success('artemisApp.studentExams.assessUnsubmittedStudentExamsSuccess', { number: res?.body });
+                    this.existsUnsubmittedExercises = false;
+                    this.isAssessingUnsubmittedExams = false;
+                },
+                error: (error: HttpErrorResponse) => {
+                    this.dialogErrorSource.next(error.message);
+                    this.alertService.error('artemisApp.studentExams.assessUnsubmittedStudentExamsFailure');
+                    this.isAssessingUnsubmittedExams = false;
+                },
+            });
+        } else {
+            this.alertService.error('artemisApp.studentExams.assessUnsubmittedStudentExamsFailure');
+        }
+    }
+
+    calculateIsExamOver() {
+        if (this.longestWorkingTime && this.exam) {
+            const startDate = dayjs(this.exam.startDate);
```

Comment:
Why do you need the conversion here?

Reply:
You mean the dayjs conversion?

Reply:
Yes, isn't this enough?
```suggestion
            const startDate = this.exam.startDate;
```

Reply:
I got this method directly from exam-assessment-buttons.ts component so I didn't write this, but I'm guessing the motivation is to not make startDate undefined